src/overlays/actors/ovl_Door_Ana/z_door_ana.c

```diff
@@ -22,19 +33,184 @@ const ActorInit Door_Ana_InitVars =
+static void func_80994124(DoorAna* this, GlobalContext* globalCtx)
+{
+    Player* player;
+    s32 destinationIdx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Ana/func_809942D8.s")
+    player = PLAYER;
+    if (Math_ApproxF(&this->actor.scale.x, 0.01f, 0.001f) != 0)
+    {
+        if ((this->actor.unk_1F != 0) && (globalCtx->sceneLoadFlag == 0) && 
+            (player->stateFlags1 & 0x80000000) && (player->unk_84F == 0))
+        {
+            destinationIdx = ((this->actor.params >> 0xC) & 7) - 1;
+            func_800C0AF4(globalCtx, 1, 0x4FF);
+            gSaveContext.respawn[RESPAWN_MODE_RETURN].pos.y = this->actor.posRot.pos.y;
+            gSaveContext.respawn[RESPAWN_MODE_RETURN].yaw = this->actor.initPosRot.rot.y;
+            gSaveContext.respawn[RESPAWN_MODE_RETURN].data = this->actor.params & 0xFFFF;
+            if (destinationIdx < 0)
+            {
+                destinationIdx = this->actor.initPosRot.rot.z + 1;
+            }
+            globalCtx->nextEntranceIndex = entrances[destinationIdx];
+            func_80993EF0(this, (ActorFunc)&func_809942D8);
+        }
+        else
+        {
+            if (func_8008E988(globalCtx) == 0 && !(player->stateFlags1 & 0x8800000) && 
+                this->actor.xzDistanceFromLink <= 15.0f && -50.0f <= this->actor.yDistanceFromLink && 
+                this->actor.yDistanceFromLink <= 15.0f)
+            {
+                player->stateFlags1 |= 0x80000000;
+                this->actor.unk_1F = 1;
+            }
+            else
+            {
+                this->actor.unk_1F = 0;
+            }
+        }
+    }
+    Actor_SetScale(&this->actor, this->actor.scale.x);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Ana/DoorAna_Update.s")
+// update function for after the player has triggered the grotto
+static void func_809942D8(DoorAna* this, GlobalContext* globalCtx)
+{
+    Player* player;
+
+    if (this->actor.yDistanceFromLink <= 0.0f && 15.0f < this->actor.xzDistanceFromLink)
+    {
+        player = PLAYER;
+        player->actor.posRot.pos.x = Math_Sins(this->actor.rotTowardsLinkY) * 15.0f + this->actor.posRot.pos.x;
+        player->actor.posRot.pos.z = Math_Coss(this->actor.rotTowardsLinkY) * 15.0f + this->actor.posRot.pos.z;
+    }
+}
+
+static void DoorAna_Update(DoorAna* this, GlobalContext* globalCtx)
+{
+    this->actionFunc(this, globalCtx);
+    // changes the grottos facing angle based on camera angle
+    this->actor.shape.rot.y = func_8005A9F4(globalCtx->cameraCtx.activeCameraPtrs[globalCtx->cameraCtx.unk_5C0]) + 0x8000;
+}
+
+static void DoorAna_Draw(DoorAna* this, GlobalContext* globalCtx)
+{
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    Gfx** dList = &D_05001390; // required for stack placement?
+    Gfx* gfxArr[3];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Ana/DoorAna_Draw.s")
+    func_800C6AC4(gfxArr, globalCtx->state.gfxCtx, "../z_door_ana.c", 0x1B8);
```

Comment:
Any time you see a function that has the file name in it like this, the last arg is a line number. Put them in decimal. Theres a couple more below